core/src/main/java/org/geysermc/geyser/command/defaults/ReloadCommand.java

```diff
@@ -52,7 +54,8 @@ public void execute(GeyserSession session, GeyserCommandSource sender, String[]
         sender.sendMessage(message);
 
         geyser.getSessionManager().disconnectAll("geyser.commands.reload.kick");
-        geyser.reload();
+        // Without the tiny wait, players do not get kicked
```

Comment:
Can you elaborate (in the code) on why this behavior changes things?

Reply:
Added a fixme; as for why, i dont really know - in my testing, session.disconnect was called for all sessions through disconnectAll(); but that never really disconnected players.